### PR TITLE
Park form setup

### PR DESCRIPTION
**Pull request by LeslieWilson**
New park form with tests and serializers

### CODE REVIEW

File: app/controllers/api/v1/parks_controller.rb [hunk 1]
```diff
@@ -1,7 +1,4 @@
 class Api::V1::ParksController < ApiController
-
-  before_action :authorize_user, except: [:index, :show]
```
Comment (CassandraNewell) [inline app/controllers/api/v1/parks_controller.rb]: Why did you guys decide to remove this authorization?

File: app/javascript/react/components/App.js [hunk 2]
```diff
@@ -2,14 +2,15 @@ import React from 'react'
...
 import IndexPage from "../containers/IndexPage"
 import ParkShowContainer from "../containers/ParkShowContainer"
-
+import ParkForm from "../containers/ParkForm"
 
 const App = (props) => {
   return(
     <BrowserRouter>
       <Switch>
         <Route exact path="/" component = {IndexPage}/>
```
Comment (CassandraNewell) [inline app/javascript/react/components/App.js]: Please remove the spaces around the `=` in each `component = ` line

File: app/javascript/react/containers/ParkForm.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    })
+    .then((persistedData)=>{
+      setShouldRedirect(true)
+    })
+    .catch((error) => {console.error("error in fetch")
+    })
+  }
+
+  if (shouldRedirect){
+    return <Redirect to="/parks" />}
```
Comment (CassandraNewell) [inline app/javascript/react/containers/ParkForm.js]: } should be on its own line

File: config/environments/development.rb [hunk 4]
```diff
@@ -14,6 +14,7 @@
   # Show full error reports.
   config.consider_all_requests_local = true
 
+  config.assets.debug = true
```
Comment (CassandraNewell) [inline config/environments/development.rb]: please switch this to `false` to improve performance

File: config/routes.rb [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
   namespace :api do
-    namespace :v1 do
-      resources :parks, only: [:index, :show]
-    end
-  end
+      namespace :v1 do
+          resources :parks, only: [:index, :show, :new, :create]
+      end
```
Comment (CassandraNewell) [inline config/routes.rb]: Please remove the extra indentation added here

File: db/schema.rb [hunk 6]
```diff
@@ -20,8 +20,6 @@
     t.string "location", null: false
     t.text "description", null: false
     t.string "image", null: false
-    t.datetime "created_at", null: false
-    t.datetime "updated_at", null: false
```
Comment (CassandraNewell) [inline db/schema.rb]: Why did you decide to remove the timestamps for parks? In general, everything should have timestamps unless there's a very good reason to omit them.

File: app/javascript/react/containers/IndexPage.js [hunk 7]
```diff
@@ -1,5 +1,8 @@
+
 import React, { useState, useEffect } from "react"
+import { BrowserRouter, Route, Switch } from "react-router-dom"
```
Comment (CassandraNewell) [inline app/javascript/react/containers/IndexPage.js]: Where are you using these new imports?

File: app/javascript/react/containers/IndexPage.js [hunk 8]
```diff
@@ -1,5 +1,8 @@
+
```
Comment (CassandraNewell) [inline app/javascript/react/containers/IndexPage.js]: Please remove this extra line

File: app/controllers/api/v1/parks_controller.rb (hunk 1 above)
Comment (CassandraNewell) [inline app/controllers/api/v1/parks_controller.rb]: Why did you decide to remove this?

File: app/javascript/react/containers/ParkForm.js [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+      } else {
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error);
+      }
+    })
+    .then((response)=>{
+      return response.json()
+    })
+    .then((persistedData)=>{
```
Comment (CassandraNewell) [inline app/javascript/react/containers/ParkForm.js]: Just a head's up -- right now, even if your controller returns `{ status: "error" }`, you'll still redirect to the index page. This is okay if it's intentional

File: app/javascript/react/components/App.js (hunk 2 above)
Comment (CassandraNewell) [inline app/javascript/react/components/App.js]: Please remove the spacing around `=` (change `component = {` to `component={`) in every route

File: app/javascript/react/containers/ErrorList.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import _ from 'lodash'
+
+const ErrorList = props => {
+  const errantFields  = Object.keys(props.errors)
+  if (errantFields.length > 0) {
```
Comment (CassandraNewell) [inline app/javascript/react/containers/ErrorList.js]: The callback function in `.map` can actually take a second argument, which represents the index. So, you can replace lines 7-9 with 
```
const listItems = errantFields.map((field, index) => {
```
which is pretty nifty.

File: app/controllers/api/v1/parks_controller.rb (hunk 1 above)
Comment (DanAlberts) [reply app/controllers/api/v1/parks_controller.rb]: user authorization isnt fully setup yet, but we will re-add it soon

File: db/schema.rb (hunk 6 above)
Comment (DanAlberts) [reply db/schema.rb]: was not intentional, we dont even remember doing this merge conflict... this is strange

Comment (DanAlberts) [reply db/schema.rb]: it is in our table which we re-migrated, it still isnt showing in our schema...

File: app/controllers/api/v1/parks_controller.rb (hunk 1 above)
Comment (DanAlberts) [reply app/controllers/api/v1/parks_controller.rb]: adding this back in broke the tests

**Pull request merged**